app/javascript/components/signup_form.tsx

```diff
@@ -0,0 +1,100 @@
+import * as React from "react";
+import axios from "axios";
+
+interface SignupProps {
+  path: string;
+  token: string;
+  redirectPath: string;
+}
+
+const SignupForm: React.FC<SignupProps> = ({ path, token, redirectPath }) => {
+  const [name, setName] = React.useState("");
+  const [sname, setSname] = React.useState("");
+  const [email, setEmail] = React.useState("");
+  const [password, setPassword] = React.useState("");
+  const [confirmpassword, setConfirm] = React.useState("");
+  const [errorsList, setErrors] = React.useState([]);
+
+  const SendForm = (e: any) => {
+    e.preventDefault();
+    axios({
+      method: "post",
+      url: path,
+      headers: {
+        "X-CSRF-Token": token
+      },
+      data: {
+        first_name: name,
+        second_name: sname,
+        email: email,
+        password: password,
+        password_confirmation: confirmpassword
+      }
+    })
+      .then((response) => {
+        if (response.status === 200 && response.data.errors === undefined) {
+          window.location.href = redirectPath;
+        }
+        setErrors(response.data.errors);
+      })
+      .catch(() => {
+        window.alert("Please try again");
+      });
+  };
+
+  return (
+    <div>
+      <form className="form" onSubmit={SendForm}>
+        <h1>SIGN-UP</h1> <br></br>
+        {errorsList
+          ? (
+          <ul className="signup-ul" data-testid="error-div">
+            {errorsList
```

Comment:
Do you still need to check for the errorslist here? You're already doing it on line 49

Reply:
My bad, overlooked this, fixed in the latest push